sdk/metric/sdk.go

```diff
@@ -117,15 +97,22 @@ type (
 		// supports checking for no updates during a round.
 		collectedCount int64
 
-		// labels is the processed label set for this record.
+		// storage is the stored label set for this record,
+		// except in cases where a label set is shared due to
+		// batch recording.
 		//
 		// labels has to be aligned for 64-bit atomic operations.
-		labels labels
+		storage label.Set
+
+		// labels is the processed label set for this record.
+		// this may refer to the `storage` field if this label
+		// set is not shared.
```

Comment:
This may also refer to the `storage` field when this label set _is_ shared, but this record was the first in batch. I'd rephrase it like so:
```suggestion
		// This may refer to the `storage` field in another record
		// if this label set is shared.
```